Orange/widgets/unsupervised/owcorrespondence.py

```diff
@@ -113,12 +115,14 @@ def set_data(self, data):
         if data is not None:
             self.varlist[:] = [var for var in data.domain.variables
                                if var.is_discrete]
-            self.selected_var_indices = [0, 1][:len(self.varlist)]
-            self.component_x = 0
-            self.component_y = int(len(self.varlist[self.selected_var_indices[-1]].values) > 1)
-            self.openContext(data)
-            self._restore_selection()
-#             self._invalidate()
+            if not len(self.varlist[:]):
+                self.Error.no_disc_vars()
```

Comment:
Add `self.data = None` here. Not needed right now, but the above code sets it to `None` as the indicator that no data is shown.